### ISSUE TITLE
Slow to open a Go file with long line

### ISSUE THREAD

**Issue by egonelbre** [jeff-hykin/better-go-syntax#2]
The grammar here has similar issues as mentioned in https://github.com/microsoft/vscode/issues/86703.

An example file: https://gist.githubusercontent.com/egonelbre/f93be4d31e7f4dad766a5e8f63374bb3/raw/747cf854bef769e3b8f57edbe207e4e001d98421/example.go

The culprit patterns:
* https://github.com/jeff-hykin/better-go-syntax/blob/54ff898316f8647d77ffcf83880a9556445326f1/source/languages/go/syntax.tmLanguage.json#L234
* https://github.com/jeff-hykin/better-go-syntax/blob/54ff898316f8647d77ffcf83880a9556445326f1/source/languages/go/syntax.tmLanguage.json#L133

By removing them, opening time for the scanning time goes from 15s to 0.006s.

**Comment by jeff-hykin** [comment]
Thanks for reporting this along with the culprit patterns, that's super helpful. I'll see what I can do in the next few weeks.

**Comment by jeff-hykin** [comment]
So the issue with the first one `(?<!var)` is because it has both a negative lookbehind and a lookahead, which are computationally inefficient.

I believe it is looking for `a_var_1, var_2.attr, a_var_3  = ` that doesn't have `var` infront of it. However I believe it also fails at checking for `var` and instead there's probably a bug whenever a variable starts with the letters `var`. This pattern probably needs to be rewritten completely, but that also means I need to do a lot of testing which will take time.

The second pattern isn't so inefficient or broken, but doesn't make very much sense and could use rewriting.

The best way to fix this is to write a new pattern for function calls, function declarations, and multiple variable assignment. But thats going to take more time.

**Comment by egonelbre** [comment]
Yeah, I have no clear idea what that is supposed to match, maybe the history books know the answer.

The second pattern I made the change `"(\\bfunc\\b)|(\\w+)(?=\\()"`  -> `"\\bfunc\\b"` and the parse time went from `3.6s -> 0.006s`. That to me seems significant, however it is tolerable compared to the first one.

I currently have already removed the patterns and live without them, as responsiveness is more important to me than correct coloring everywhere. Unfortunately I'm not sure when do I get around to fixing it, but if I do find some time, I'll make a PR.

**Comment by egonelbre** [comment]
It seems the first one corresponds to https://github.com/atom/language-go/blob/master/grammars/go.cson#L206

**Comment by egonelbre** [comment]
I just realized that maybe instead of negatively matching on `var`, it could match on `[{;\n][ \t]\s*`.

**Comment by jeff-hykin** [comment]
> I just realized that maybe instead of negatively matching on `var`, it could match on `[{;\n][ \t]\s*`. Since it is a statement the only valid place would be on a completely new line, after `;` , or after an opening brace.

That is helpful to know. This kind of assumption is what I use on the C++ syntax, and its where my Go Lang knowledge is limited.

**Comment by jeff-hykin** [comment]
I added a `go (simplified)` language and published it with the `Better Go Syntax` extension. So now you can install the extension and configure '.go' files to open with "go (simplified)" to prevent the slowdown. The simplified version is just the syntax with those patterns removed

**Comment by jeff-hykin** [comment]
> Yeah, I have no clear idea what that is supposed to match, maybe the history books know the answer.

Most of the WTF parts of TM syntaxes I have read/changed are 80% just broken and 20% are intentional edgecase behavior
